workspaces/src/result.rs

```diff
@@ -105,15 +119,18 @@ impl CallExecutionDetails {
         }
     }
 
-    /// Convert the execution details into a Result if its status is not a successful one.
-    /// Useful for checking if the call was successful and forwarding the error upwards.
-    fn try_into_result(self) -> Result<Self> {
+    /// Checks whether the transaction was successful, if not returns a Result with the
+    /// corresponding error that occured.
+    pub fn ok(&self) -> Result<()> {
```

Comment:
This is a very strange pattern, the goal is to require the developer to use these `ok()` functions to check for errors and retrieve the inner values through another call? This feels very janky

Reply:
The intent of this call was for them to call into `ok()` so that they don't trigger the `#[must_use]` warning while ignoring the rest of the execution details. Now, if they want to use the execution details, they would normally just reference all the other details instead of using `ok()`. Like if they just want to grab the details of the execution:
```rust
let details = contract.call(...)
   .transact()
   .await?;

println!("gas burnt: {}", details.total_gas_burnt);
```
or if they don't care at all about the details, but just want to know if the execution succeeded or not:
```rust
contract.call(...)
   .transact()
   .await?
   .ok()?;
```
It's just one way or the other, but not both at the same time, since we already have functions like `json`/`borsh` that allows us to deal with the error similarly if that makes sense.

Might not be the best way of expressing that though.